test/MaeParserTest.cpp

```diff
@@ -587,7 +587,7 @@ BOOST_AUTO_TEST_CASE(String)
         Buffer b(ss);
         whitespace(b);
         std::string s = parse_value<std::string>(b);
-        BOOST_REQUIRE_EQUAL(s, "Q\\ Z");
+        BOOST_REQUIRE_EQUAL(s, R"(Q\ Z)");
```

Comment:
If I load this string into maestro, it will be `Q Z`, not `Q\ Z`, so it looks like the mmshare implementation allows escaping any characters, not only `\` and `"`.

Reply:
Ugh, but this makes no sense.  A protected b isn't just 'b', [it has a meaning](https://stackoverflow.com/questions/10220401/rules-for-c-string-literals-escape-character), if we strip the protection we're just ignoring that.  Even if they didn't have a meaning, The obvious example of that being true is the protected 'n', which should be interpreted as newline by whatever reads it.

I'm hesitating on trying to perfectly reproduce Maestro's behavior on edge cases where it behaves poorly.  I doubt we rely on this anywhere in Maestro.

Reply:
I can't disagree, but I think our reasoning is at a tie. You want to treat `"\n"` like `"\\n"`, I suggested to treat it like `"n"`, both cases have lost the original information. I still feel like I win because Maestro is on my side ;-)

Reply:
Yeah, that's fair enough.  And if we're going to allow both parsers to exist in mmshare at once (which we're planning on doing), then I suppose they have to get as close to the exact same as possible - so it's either change m2io or change this.  I'll change this to just strip '\' naively.